src/nettyrpc.nim

```diff
@@ -1,49 +1,121 @@
-import std/[macros, strutils, macrocache]
+import std/[macros, macrocache, tables, hashes]
 import netty
 import nettyrpc/nettystream
 
 export nettystream
 
 type NettyRpcException = object of CatchableError
 
-var 
+var
   compEventCount{.compileTime.} = 0u16
-  events*: array[uint16, proc(data: var NettyStream)] ## Ugly method of holding procedures
+  relayedEvents*: array[uint16, proc(data: var NettyStream, conn: Connection)] ## Ugly method of holding procedures
+  managedEvents*: Table[Hash, proc(data: var NettyStream, conn: Connection)] ## Uglier method of holding procedures
   reactor*: Reactor
   client*: Connection
   sendBuffer* = NettyStream()
 
-proc send*(message: var NettyStream) =
-  ## Sends the RPC message to the server to relay
+proc basicSend(conn: Connection, message: string) =
+  ## Sends the RPC message to the server to process directly.
+  ## Does not use netty stream.
+  if reactor.isNil:
+    raise newException(NettyRpcException, "Reactor is not set")
+  if(not reactor.isNil):
+    reactor.send(conn, message)
+
+proc send*(conn: Connection, message: var NettyStream) =
+  ## Sends the RPC message to the server to process directly.
+  if reactor.isNil:
+    raise newException(NettyRpcException, "Reactor is not set")
+  if(not reactor.isNil and message.pos > 0):
+    reactor.send(conn, message.getBuffer[0..<message.pos])
+    message.clear()
+
+proc sendall*(message: var NettyStream) =
+  ## Sends the RPC message to the server to process
   if reactor.isNil:
     raise newException(NettyRpcException, "Reactor is not set")
   if(not reactor.isNil and message.pos > 0):
-    reactor.send(client, message.getBuffer[0..<message.pos])
+    var d = message.getBuffer[0..<message.pos]
+    for conn in reactor.connections:
+      reactor.send(conn, d)
     message.clear()
 
-proc rpcTick*(client: Reactor) =
+proc rpc*(conn: Connection, procName: string, vargs: tuple) =
```

Comment:
For all of these rpc procedures we could utilize static strings to avoid hashing the string at runtime(saves a bit of performance) so something like the following could be utilized
```nim
type Strings* = string or static string

proc rpc*(conn: Connection, procName: Strings, vargs: tuple) =
  ## Send a rpc to a specific connection.
  when procName.type is static string:
    const hashedName = hash(procName)
    sendBuffer.write(hashedName)
  else:
    sendBuffer.write(hash(procName))
  for k, v in vargs.fieldPairs:
    sendBuffer.write(v)
  send(conn, sendBuffer)
  ```

Reply:
Ok, will get this implemented.  Can I continue to commit or do I need to put in a new PR

Reply:
You can continue on this PR

Reply:
@beef331 where do you get the `Strings` type?

Reply:
Oh nvm I'm blind haha